nixos/modules/services/web-apps/pixelfed.nix

```diff
@@ -0,0 +1,366 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.pixelfed;
+  opt = options.services.pixelfed;
+
+  user = "pixelfed";
+  group = "nginx";
+
+  pixelfed = pkgs.pixelfed.override {
+    dataDir = cfg.dataDir;
+  };
+
+  configFile = pkgs.writeTextFile {
```

Comment:
this config file is going to end up in the nix store and compromise the secret
IMHO:

- either we provide a file loading mechanism at runtime to load securely the secrets
- either we do put up some documentation saying this is not secure

can help you with that if needed